src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -191,9 +191,7 @@ public function set($id, $service, $scope = self::SCOPE_CONTAINER)
             $this->scopedServices[$scope][$id] = $service;
         }
 
-        if (isset($this->aliases[$id])) {
-            unset($this->aliases[$id]);
-        }
+        unset($this->aliases[$id], $this->methodMap[$id]);
```

Comment:
:+1:, even if it won't have any practical effect because `$this->services[$id]` is checked first in the get() method